### ISSUE TITLE
GitHub Actions CI?

### ISSUE THREAD

**Issue by jackfirth** [florence/cover-codecov#2]
I wrote [this GitHub Action][action] for building and testing Racket packages. How would I go about using the `build-services` info key to support using `raco cover` in that environment?

[action]: https://github.com/jackfirth/racket-package-ci-action

**Comment by florence** [comment]
yeah I should get around to making actually documentation for that someday....

For now, follow the example set out by: https://github.com/florence/cover-codecov/blob/master/cover/private/gitlab-service.rkt

to make the plugin, then add something like:

```
(define cover-build-services '(("github-service" <require-path>
                                                     github-ci?
                                                     github-service@))
```

to your info.rkt, where `github-ci?` has the type `(-> boolean?)` and is true when the github actions CI is the CI building the project. The `github-service@` should provide a function, `(query)` which gives an alist which contains the POST parameters that codecov would need to integrate with github actions.


That being said, I don't think that codecov quite supports Actions yet. See https://github.com/codecov/codecov-node/issues/118,  which I understand to mean that the POST query will need to use the generic API that gets some secret key. (I honestly don't think I ever used this part of the codecov API, so I wont be much help here. maybe @rpless knows the right want to integrate in this case?)

**Comment by jackfirth** [comment]
There's now an [action][codecov-action] specifically for uploading just the coverage XML file to Codecov. Can cover generate that XML file?

[codecov-action]: https://github.com/codecov/codecov-action
